### ISSUE TITLE
Event types do not exist in the Window interface

### ISSUE THREAD

**Issue by kevipapo1** [Microsoft/TypeScript#19758]
**TypeScript Version:**  2.7.0-dev.201xxxxx

**Code**

```shell
npm install typescript jsdom @types/jsdom
```

```ts
// example.ts

import { JSDOM } from 'jsdom';

let dom = new JSDOM(`
    <!DOCTYPE html>
    <html>
        <head>
            <title>Test DOM</title>
        </head>
        <body>
            <input type="button" id="button" />
        </body>
    </html>
`);
let window = dom.window;
let elem = window.document.getElementById('button');
elem.onclick = (ev: MouseEvent) => {
    console.log(`Button clicked, typeArg: ${ev.type}`);
}
let event: MouseEvent = new window.MouseEvent('someTypeArg');
elem.onclick(event);
```

**Expected behavior:**

TypeScript compiles, running `node example.js` outputs the following to console, then exits:

```
Button clicked, typeArg: someTypeArg
```

**Actual behavior:**

TypeScript fails to compile with the following error:

```
example.ts(19,36): error TS2339: Property 'MouseEvent' does not exist on type DOMWindow
```

--------------------------------------

[@types/jsdom](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jsdom/index.d.ts) defines the `DOMWindow` interface as:

```ts
export interface DOMWindow extends Window { eval(script: string): void; }
```

The `Window` interface it extends is the one defined by TypeScript. That interface is missing definitions for the Event types, although these Event types are declared outside the `Window` interface. In web browsers, the Event types exist both globally and on the `window` object. TypeScript currently handles the former but not the latter.

[jsdom](https://github.com/tmpvar/jsdom) does in fact implement the Event types on the `dom.window` object. You can modify the code above to compile and behave as expected by extending the `DOMWindow` interface, adding the `MouseEvent` type to it, and casting the `dom.window` object to this newly defined interface:

```ts
import { JSDOM, DOMWindow } from 'jsdom';

interface WindowWithMouseEvent extends DOMWindow {
    MouseEvent: {
        prototype: MouseEvent;
        new(typeArg: string, eventInitDict?: MouseEventInit): MouseEvent;
    }
}

let dom = new JSDOM(`
    <!DOCTYPE html>
    <html>
        <head>
            <title>Test DOM</title>
        </head>
        <body>
            <input type="button" id="button" />
        </body>
    </html>
`);
let window = dom.window as WindowWithMouseEvent;
let elem = window.document.getElementById('button');
elem.onclick = (ev: MouseEvent) => {
    console.log(`Button clicked, typeArg: ${ev.type}`);
}
let event: MouseEvent = new window.MouseEvent('someTypeArg');
elem.onclick(event);
```

I've spoken to the maintainer of [@types/jsdom](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jsdom/index.d.ts) (see [this issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/21235)) and they suggested raising an issue with your team instead. I agree, seeing as Event types exist on the `window` object in web browsers.

**Comment by aluanhaddad** [comment]
Just FYI, a simpler workaround is to write
```ts
declare global {
  interface Window {
    MouseEvent: typeof MouseEvent;
  }
}
```

**Comment by kevipapo1** [comment]
If you'd like, I can submit a PR for this

**Comment by mhegazy** [comment]
To model the spec/runtime behavior correctly, `window` should be an alias to `global`. In other words all declarations that are visible in the global namespace, should be accessible through `window`. This applies to things like `JSON`, `Math`, `HTMLElement`, `URLSearchParams`, and `MouseEvent`.

We have chosen a while back to not expose these types on Window. We have added ones thought to the interface, but i am afraid there is not really a clear pattern/principal here.

See https://github.com/Microsoft/TypeScript/issues/18756 for related discussion.

I think we should revisit the original issue of making `Window` the global polluter and be honest to the spec. I think https://github.com/Microsoft/TypeScript/issues/14052 is a prerequisite here, but there could be other proposals too.

I have created https://github.com/Microsoft/TypeScript/issues/19816 to track general issue.
